docs: Navbar doc fixes

dwgray opened the pull request:
# Fixes two issues with the NavBar docs

1. The use of type="light|dark" has been replaced with data-bs-theme="light|dark" in order to modify the text color of the NavBar.  I've replaced all cases in the samples and updated the description.  This addresses #1507 and makes the samples readable again.
2. Changed the classes in the initial sample to make the comment stating that a portion of the navbar would be right aligned accurate.  This was a matter of changing me-auto to ms-auto to conform with bootstrap 5's utility classes.  This addresses #1510 

## Small replication

Fixes #1507 and #1510

## PR checklist

<!-- (Update "[ ]" to "[x]" to check a box) -->

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix :bug: - `fix(...)`
- [ ] Feature - `feat(...)`
- [ ] ARIA accessibility - `fix(...)`
- [x] Documentation update - `docs(...)`
- [ ] Other (please describe)

**The PR fulfills these requirements:**

- [x] Pull request title and all commits follow the [**Conventional Commits**](https://www.conventionalcommits.org/) convention or has an [**override**](https://github.com/googleapis/release-please#how-can-i-fix-release-notes) in this pull request body **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type. Pull requests that do not follow conventional commits or do not have an override will be denied**

VividLemon commented:
Use the color mode options rather than the raw data-bs-theme attribute. Then make a section about it, if there isn't one already, discussing all three options -- directive, conposable, and raw attribute

dwgray commented:
Okay, I looked into the options you talked about in a little more detail and I still think the primary use of overriding the text color is to use the raw attribute to override the underlying theme text color when the variant is set so that changing the light/dark theme doesn't make the text in the navbar unreadable.  This exactly what the docs are doing with the navbar - the custom primary color is dark enough that you need to set the text color to the dark theme color to make it readable.  So I changed the first sample to match that use case.

I added a line to note that the navbar conforms to theme colors if the variant property isn't set.  This seems to be the place where useColorMode is relevant.

The v-b-color-mode directive seems to only take a variable, not an constant.  So it doesn't seem to be a solution to the problem that is being addressed here.  I guess there is a case where one would programmatically change both the variant (not using themes), but that seems like a very corner case.  Let me know if I'm missing something here.

VividLemon commented:
/
![Uploading Screenshot_20231005-171008.png…]()

VividLemon commented:
![Screenshot_20231005-171008](https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/31498096/5ddf8edc-c237-4e13-951f-a947b314cc82)

dwgray commented:
Okay, done.  Not sure how I misread the code in the Layout.vue file, but I obviously did.

VividLemon commented on apps/docs/src/docs/components/navbar.md:
````diff
@@ -87,9 +87,25 @@ The component `BNavbar` is a wrapper that positions branding, navigation, and ot
+
+Use the `v-b-color-mode` directive:
+
+```vue-html
````
Code blocks need to be in highlight cards

dwgray replied:
I believe this is now addressed.

VividLemon commented on apps/docs/src/docs/components/navbar.md:
```diff
@@ -91,20 +91,40 @@ The component `BNavbar` is a wrapper that positions branding, navigation, and ot
 
 If the `variant` property is set, it may be necessary to control the text color so that it contrasts with the variant
-irrespective of the theme. This can be done in one of two ways.
+irrespective of the theme. This can be done by use the `v-b-color-mode` directive or the `data-bs-theme` attribute.
```
This doesn't reference `useColorMode`

dwgray replied:
I believe this is addressed by `BNavbar` will conform to the current color theme if the `variant` prop is not set. If you'd like me to explicitly include how you set the current theme and put a link to useColorMode I will.

VividLemon commented on apps/docs/src/docs/components/navbar.md:
````diff
@@ -91,20 +91,40 @@ The component `BNavbar` is a wrapper that positions branding, navigation, and ot
 ```
 
-Or use the raw `data-bs-theme` attribute:
+</template>
````
Indent

dwgray replied:
This should now be fixed.

VividLemon commented on apps/docs/src/docs/components/navbar.md:
````diff
@@ -91,20 +91,40 @@ The component `BNavbar` is a wrapper that positions branding, navigation, and ot
 ```vue-html
-<BNavbar toggleable="lg" variant="primary" v-b-color-mode="'dark'">
+<!-- Using the v-b-color-mode directive -->
+<BNavbar variant="faded">
````
The example for the template needs to match the codeblock from above

VividLemon replied:
The comment is unnecessary. But perhaps you can label it actually above each `<HighlightCard>` above

```html
With `v-b-colorMode` more info [here](#pathToVBColorMode)

<HighlightCard>
...
```

etc etc for each example

dwgray replied:
I've moved these to text above the cards with links.

VividLemon commented on apps/docs/src/docs/components/navbar.md:
````diff
@@ -91,20 +91,40 @@ The component `BNavbar` is a wrapper that positions branding, navigation, and ot
 ```vue-html
-<BNavbar toggleable="lg" variant="primary" data-bs-theme="dark">
+<!-- Using the data-bs-theme attribute -->
+<BNavbar data-bs-theme="dark">
````
The example for the template needs to match the codeblock from above

dwgray replied:
This is now fixed

dwgray commented:
Sorry about that last one, just plain sloppy on my part.  

On the consistency issue - do you have any thoughts on tools to assist or enforce that consistency between the codeblock and the template?  I can think of a number of different things that _might_ improve this, but don't want to jump into anything if you've already got thoughts...

VividLemon commented:
> if you've already got thoughts...

I have no suggestions

dwgray commented:
On the template issue, I imagine that it's not that high a priority because once you get into the workflow of editing the docs it's probably easy enough to get used to doing a copy-paste of one to the other rather than editing them individually (which is what I'll do going forward). I'll think about it a bit more and if I think it's worth pursuing I'll post.  Would that be most appropriate on the discussion section or as an issue?

dwgray commented:
Okay, I believe I've addressed all of the comments - and hopefully the last commit with this indenting covered what you were asking about and didn't cause new problems :-)  Sorry for the amount of back and forth on this - new process for me and I use Azure Devops rather than Git in my day job, so that caused a few hiccups.  Thanks for your patience.  Let me know if you need anything else from me to get this in.

VividLemon commented on apps/docs/src/docs/components/navbar.md:
```diff
@@ -87,9 +87,49 @@ The component `BNavbar` is a wrapper that positions branding, navigation, and ot
+following values to change the background color: primary, secondary, success, info, warning, danger, dark, light, or any of the \*-subtle variants.
 
-Control the text color by setting `type` prop to `light` for use with light background color variants, or dark for dark background color variants.
+If the `variant` property is set, it may be necessary to control the text color so that it contrasts with the variant
```
```suggestion
If the `variant` prop is set, it may be necessary to control the text color so that it contrasts with the variant
```

VividLemon commented on apps/docs/src/docs/components/navbar.md:
```diff
@@ -87,9 +87,49 @@ The component `BNavbar` is a wrapper that positions branding, navigation, and ot
 
-Control the text color by setting `type` prop to `light` for use with light background color variants, or dark for dark background color variants.
+If the `variant` property is set, it may be necessary to control the text color so that it contrasts with the variant
+irrespective of the theme. This can be done by use the `v-b-color-mode` directive or the `data-bs-theme` attribute.
```
```suggestion
irrespective of the theme. This can be done by use the `v-b-color-mode` directive or `useColorMode` composable
```

VividLemon commented on apps/docs/src/docs/components/navbar.md:
````diff
@@ -87,9 +87,49 @@ The component `BNavbar` is a wrapper that positions branding, navigation, and ot
+
+  </template>
+</HighlightCard>
+
+An example using `data-bs-theme`. For more information see the
+[bootstrap documentation](https://getbootstrap.com/docs/5.3/components/navbar/#color-schemes).
+
+<HighlightCard>
+  <BNavbar variant="primary" data-bs-theme="dark">
+    <BNavbarBrand tag="h1" class="mb-0">BootstrapVue</BNavbarBrand>
+  </BNavbar>
+  <template #html>
+
+```vue-html
+<BNavbar variant="primary" data-bs-theme="dark">
+  <BNavbarBrand tag="h1" class="mb-0">BootstrapVue</BNavbarBrand>
+</BNavbar>
+```
+
+  </template>
+</HighlightCard>
+
````
```suggestion
```

VividLemon commented on apps/docs/src/docs/components/navbar.md:
```diff
@@ -87,9 +87,30 @@ The component `BNavbar` is a wrapper that positions branding, navigation, and ot
+If the `variant` prop is set, it may be necessary to control the text color so that it contrasts with the variant
+irrespective of the theme. This can be done by use the `v-b-color-mode` directive or `useColorMode` composable
+
+An example using `v-b-colorMode`. Form more information see the [BColorMode directive](/docs/directives/BColorMode).
```
```suggestion
An example using `v-b-colorMode`. For more information see the [BColorMode directive](/docs/directives/BColorMode).
```

VividLemon commented on apps/docs/src/docs/components/navbar.md:
````diff
@@ -87,9 +87,30 @@ The component `BNavbar` is a wrapper that positions branding, navigation, and ot
+```
+
+  </template>
+</HighlightCard>
````
```suggestion
</HighlightCard>

```
